### ISSUE TITLE
get ubuntu.2204.armarch[.open] deployments working

### ISSUE THREAD

**Issue by dougbu** [dotnet/dnceng#813]
addressing #777 by reverting the `cryptography` version bump (in [!33613](https://dev.azure.com/dnceng/internal/_git/dotnet-helix-machines/pullrequest/33613)) in [dotnet-helix-machines-ci](https://dev.azure.com/dnceng/internal/_build?definitionId=596&_a=summary) apparently did not fix problems deploying the ubuntu.2204.armarch[.open] scale sets. problem is most obvious at the moment (due to masking issues in main) in the latest build for [!33609](https://dev.azure.com/dnceng/internal/_git/dotnet-helix-machines/pullrequest/33609): [#20230914.04](https://dev.azure.com/dnceng/internal/_build/results?buildId=2267149&view=results)

<details>
<summary>Deployment error details</summary>

``` text
The handler for VM extension type 'Microsoft.Azure.Monitor.AzureMonitorLinuxAgent' has reported terminal failure for VM extension 'Microsoft.Azure.Monitor.AzureMonitorLinuxAgent' with error message: '[ExtensionOperationError] Non-zero exit code: 51, /var/lib/waagent/Microsoft.Azure.Monitor.AzureMonitorLinuxAgent-1.27.5/./shim.sh -install

[stdout]
2023/09/14 12:14:43 [Microsoft.Azure.Monitor.AzureMonitorLinuxAgent-1.27.5] Install,failed,51,Unsupported operating system: ubuntu 22.04

[stderr]
/var/lib/waagent/Microsoft.Azure.Monitor.AzureMonitorLinuxAgent-1.27.5/./agent.py:52: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives
  from distutils.version import LooseVersion
'.
    
'Install handler failed for the extension. More information on troubleshooting is available at https://aka.ms/vmextensionlinuxtroubleshoot' (Code: VMExtensionHandlerNonTransientError, Target: 1)
```
</details>

### Release Note Category

- [ ] Feature changes/additions 
- [x] Bug fixes
- [ ] Internal Infrastructure Improvements

### Release Note Description

Removed `AzSecPack` from the image for the `ubuntu.2204.armarch[,open]` queues and made them available (not deadlettered) again. Feel free to migrate test jobs to the new queue. This is particularly recommended for Docker testing that requires an ARM64 VM.

**Comment by dougbu** [comment]
dug a bit and found https://learn.microsoft.com/en-us/azure/azure-monitor/agents/agents-overview#linux the table there indicates Ubuntu 22.04 ARM64 machines are not yet supported.

it's not clear to me why the extension is installed. we don't do anything explicit about the `AzureMonitorLinuxAgent` extension in helix-machines and automated processes should be smart enough to avoid unsupported extensions.

bottom line, we should hunt more to find a way to disable this extension

/cc @AlitzelMendez since you added the ubuntu.2204.armarch queues
also /cc @ilyas1974 and @garath in case there's something hidden behind a string other than `AzureMonitorLinuxAgent` that enables this extension.

**Comment by dougbu** [comment]
opened [ticket 2309220010002513](https://ms.portal.azure.com/#view/Microsoft_Azure_Support/SupportRequestDetails.ReactView/id/%2Fsubscriptions%2Fcab65fc3-d077-467d-931f-3932eabf36d3%2Fproviders%2FMicrosoft.Support%2FsupportTickets%2Fe9e31931-38b75c38-e078f596-59c6-47d7-8f21-30965f5350ec) about this

**Comment by dougbu** [comment]
we have temporarily worked around this issue.

however [!33909](https://dev.azure.com/dnceng/internal/_git/dotnet-helix-machines/pullrequest/33909) was only a short term mitigation. had some conversations w/ Azure about [ticket 2309220010002513](https://ms.portal.azure.com/#view/Microsoft_Azure_Support/SupportRequestDetails.ReactView/id/%2Fsubscriptions%2Fcab65fc3-d077-467d-931f-3932eabf36d3%2Fproviders%2FMicrosoft.Support%2FsupportTickets%2Fe9e31931-38b75c38-e078f596-59c6-47d7-8f21-30965f5350ec) but we continue to wait for a solution or suggestion

**Comment by dougbu** [comment]
update: Azure has let us know enabling Microsoft Defender for Cloud (MDC) in a scale set will cause the offending extension's installation. no word yet on how to avoid the issue

**Comment by dougbu** [comment]
https://github.com/Azure/azure-linux-extensions/pull/1815 change should unblock us. will put up a PR&hellip;

**Comment by dougbu** [comment]
Final resolution remains unclear. I added @garath and @ilyas1974 to ticket [2309220010002513](https://ms.portal.azure.com/#view/Microsoft_Azure_Support/SupportRequestDetails.ReactView/id/%2Fsubscriptions%2Fcab65fc3-d077-467d-931f-3932eabf36d3%2Fproviders%2FMicrosoft.Support%2FsupportTickets%2Fe9e31931-38b75c38-e078f596-59c6-47d7-8f21-30965f5350ec).

Unassigning myself from this. About to create an issue and put up a related PR but won't deal w/ the meat of the problems unless I'm on Ops&hellip;

**Comment by dougbu** [comment]
see #1358 for interim status

**Comment by dougbu** [comment]
I created Ubuntu2204ArmArch in the helix-repro-vms DTL. I'm not un-claiming it immediately b/c I'm not sure that'll cause the VM to be wiped away. If anyone wants to look at it, please let me know and I'll make it available.

This VM uses the image generated in the last failed run of [!34863](https://dev.azure.com/dnceng/internal/_git/dotnet-helix-machines/pullrequest/34863). The deployment failed w/ something more like the original symptoms this time. At the moment, the failed deployment information is visible in the `Deployments` tab of [pr-ubuntu.2204.armarch-dougbu-reactivate.arma-westus2-v2-rg](https://ms.portal.azure.com/#@microsoft.onmicrosoft.com/resource/subscriptions/cab65fc3-d077-467d-931f-3932eabf36d3/resourceGroups/pr-ubuntu.2204.armarch-dougbu-reactivate.arma-westus2-v2-rg/overview) and [pr-ubuntu.2204.armarch.open-dougbu-reactivate.arma-westus2-v2-rg](https://ms.portal.azure.com/#@microsoft.onmicrosoft.com/resource/subscriptions/cab65fc3-d077-467d-931f-3932eabf36d3/resourceGroups/pr-ubuntu.2204.armarch.open-dougbu-reactivate.arma-westus2-v2-rg/overview).

The bit that was missing for a while was `Unsupported operating system: ubuntu 22.04`. The [extension history](https://github.com/Azure/azure-linux-extensions/commits/master/AzureMonitorAgent) doesn't show anything which explicitly reverts [the Ubuntu 22.04 on ArmArch change](https://github.com/Azure/azure-linux-extensions/commit/0b91dacd98e1ea63933a52d7899609956114471e) but perhaps `syslog` isn't enabled on these machines or something similar❔

Interestingly, 3 of 5 sub-deployments (`deployAddUAName-okuyfqoo4nsww` or `deployAddUAName-zggs6a5dzpuzc`, `set-template-a`, and `PrimaryDeployment`) failed this time. I think I'm used to seeing only 2/5 failing. More interestingly, `PolicyDeployment_2897317476391990803` **succeeded** and that's the one that installs the failing extension.

----

Capturing the info for posterity:
``` text
The handler for VM extension type 'Microsoft.Azure.Monitor.AzureMonitorLinuxAgent' has reported terminal failure for VM extension 'Microsoft.Azure.Monitor.AzureMonitorLinuxAgent' with error message: '[ExtensionOperationError] Non-zero exit code: 51, /var/lib/waagent/Microsoft.Azure.Monitor.AzureMonitorLinuxAgent-1.28.11/./shim.sh -install
[stdout]
2023/11/16 21:37:09 [Microsoft.Azure.Monitor.AzureMonitorLinuxAgent-1.28.11] Install,failed,51,Unsupported operating system: ubuntu 22.04

[stderr]
/var/lib/waagent/Microsoft.Azure.Monitor.AzureMonitorLinuxAgent-1.28.11/./agent.py:52: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives
  from distutils.version import LooseVersion
'.
    
'Install handler failed for the extension. More information on troubleshooting is available at https://aka.ms/vmextensionlinuxtroubleshoot' (Code: VMExtensionHandlerNonTransientError, Target: 2)
```

**Comment by dougbu** [comment]
@markwilkie, @ilyas1974, @missymessa I _think_ we decided to put this into `Operations` and not `Ops - Service Maintenance`. If I'm right, it should be moved. It also shouldn't have and assignee so that the person on Ops knows it's available to be picked up.

From my perspective, we have enough information to continue the conversation w/ Azure. However I'd rather not drive things unless someone is saying this should be part of the Helix Machines Improvements epic.

**Comment by chcosta** [comment]
I keep losing this issue, so I'm assigning it to me for now

**Comment by chcosta** [comment]
Latest suggestion for a repro from Doug, putting it here so I don't lose it too..

> suggest creating your own VM in
[HelixPRImages-DTL - Microsoft Azure](https://ms.portal.azure.com/#@microsoft.onmicrosoft.com/resource/subscriptions/84a65c9a-787d-45da-b10a-3a1cefce8060/resourceGroups/HelixPRImages/providers/Microsoft.DevTestLab/labs/HelixPRImages-DTL/overview)
. use the Standard_D2ps_v5 size and the ubuntu.2204.armarch image w/ version 2023.1104.003800. that's the image from my last failed run of
[Pull request 34863: Reactivate ubuntu.2204.armarch[.open] image - Repos (azure.com)](https://dev.azure.com/dnceng/internal/_git/dotnet-helix-machines/pullrequest/34863)
. (Image versions are found in the first dropdown of the Advanced Settings tab.) You can mess w/ extensions and so on by searching in Azure for the new VM name and finding it a longer-named resource group (it'll be in two -- one for the DTL and the "real" one that gives you more control.)

**Comment by dougbu** [comment]
fixed in [!34863](https://dev.azure.com/dnceng/internal/_git/dotnet-helix-machines/pullrequest/34863)
